### PR TITLE
Nim packages

### PR DESCRIPTION
**Pull request by ghost**
##### Motivation for this change

Repackage [NImble](https://github.com/nim-lang/nimble) packages using Nix.

This is yet another `mkDerivation` wrapper for yet another programing language and package manager.

A utility is provided at `nix-generate-from-nimble` for generating a JSON package description from Nimble metatdata and a skeleton `default.nix` file.

The first example package is an interpreter for the [Spry](http://sprylang.se/) language.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @peterhoeg @dom96

### CODE REVIEW

File: pkgs/development/nim-modules/nimble/json.patch [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+     of actionInstall:
+diff --git a/src/nimblepkg/packageinfo.nim b/src/nimblepkg/packageinfo.nim
+index c54b97d..ab02590 100644
+--- a/src/nimblepkg/packageinfo.nim
++++ b/src/nimblepkg/packageinfo.nim
+@@ -427,6 +427,17 @@ proc echoPackage*(pkg: Package) =
+     if pkg.web.len > 0:
+       echo("  website:     " & pkg.web)
+ 
++func json*(pkg: Package): JsonNode =
```
Comment (dom96) [inline pkgs/development/nim-modules/nimble/json.patch]: Might be worth contributing this upstream? :)

Comment (ehmry) [reply pkgs/development/nim-modules/nimble/json.patch]: Yes, I tried to format the patch for a clean PR, but I'd like to tinker a bit more before I do that.

**Comment by ehmry** [comment]
There are over a thousands PRs pending, not feasible in this repo.

**Comment by NickHu** [comment]
Can I just ask what is going to happen to this PR? It looks fairly complete and it would be a shame to see all that work not get merged.

**Comment by ehmry** [comment]
@NickHu it was moved to a seperate flake-style repo, but is still a bit incomplete: https://github.com/nix-community/flake-nimble

**Pull request closed without merging**